dprital/sonic-buildimage
# [Mellanox] Change NVIDIA SDK build to cmake method

#### Why I did it

Up to now, building Mellanox SDK debian packages was done by "autotools" method.
Lately, Mellanox SDK team added the support to build by "cmake" method.
The benefit of using "cmake" is with decrease time of SDK build as "cmake" make it faster then "auto build".

When moving to "cmake" method, all SDK debians:

* applibs
* sx_complib
* sx_example
* sx-gen-utils
* sx-scew
* sxd-libs
* python-sdk-api
* sx-acl-helper
* wjh-libs
* sx-hash-calc
* sx-kernel

will be replaced by single debian - **sys_sdk**.

##### Work item tracking

#### How I did it

Change SDK Makefiles, delete all Makefiles of removed debians and add new Makefile for building sys_sdk.
Change sdk.mk and sdk.dep acordingly.
sx-kernel package was remain since it is consumed by the host as well and I didn't want to consume **sys_sdk** in host.

From: _one-image.mk_
`
$(SONIC_ONE_IMAGE)_INSTALLS += $(SX_KERNEL) $(KERNEL_MFT) $(MFT_OEM) $(MFT) $(MLNX_HW_MANAGEMENT)
`

#### How to verify it



* Enter syncd docker and verify the list of SDK packages are sx-kernel and sys-sdk only
* Run regression tests and verify there is no degradation
* Test the build time when using "cmake" for building SDK packages, compare to "auto build"

#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
